### PR TITLE
feat(rest): delete a vendor by id.

### PR DESCRIPTION
**Pull request by rudra-superrr**
Issue: #2624 

Description: run the endpoint to delete a vendor by id.

### CODE REVIEW

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/vendor/VendorController.java [hunk 1]
```diff
@@ -108,6 +110,37 @@ public ResponseEntity<EntityModel<Vendor>> getVendor(
...
+    ) {
+        User sw360User = restControllerHelper.getSw360UserFromAuthentication();
+        boolean idFound = false;
+        List<Vendor> vendors = vendorService.getVendors();
+        for (Vendor vendor : vendors) {
+            if (vendor.getId().equals(id)) {
+                idFound = true;
+                break;
+            }
+        }
```
Comment (GMishx) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/vendor/VendorController.java:131]: Do we not have something like `getVendorById()` ? That would be much more efficient.

**Review by GMishx — COMMENTED** [review]
Suggestion/question

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/vendor/VendorController.java (hunk 1 above)
Comment (rudra-superrr) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/vendor/VendorController.java]: made the changes.

**Review by GMishx — APPROVED** [review]
Changes looks good.

**Comment by GMishx** [comment]
@rudra-superrr please resolve merge conflicts.

@keerthi-bl please test the PR.

**Review by GMishx — APPROVED** [review]
Approved by @keerthi-bl here: https://github.com/eclipse-sw360/sw360/pull/2626#issuecomment-2463873428

@keerthi-bl , please use "Review changes" feature of GitHub to approve/request changes on PR.

**Pull request merged**